test/camunda-cloud/ElementTemplatesReplaceProviderSpec.js

```diff
@@ -0,0 +1,313 @@
+import {
+  inject,
+  getBpmnJS,
+  bootstrapCamundaCloudModeler
+} from 'test/TestHelper';
+
+import {
+  attr as domAttr,
+  queryAll as domQueryAll,
+  query as domQuery
+} from 'min-dom';
+
+import {
+  isTemplateApplied
+} from 'lib/camunda-cloud/features/replace/ElementTemplatesReplaceProvider';
+
+import simpleXml from '../fixtures/replaceElementTemplates.bpmn';
+import templates from './replace-element-templates.json';
+
+
+describe('<ElementTemplatesReplaceProvider>', function() {
+
+  beforeEach(bootstrapCamundaCloudModeler(
+    simpleXml
+  ));
+
+  beforeEach(inject(function(elementTemplates) {
+    elementTemplates.set(templates);
+  }));
+
+
+  describe('display', function() {
+
+    it('should display matching element templates', inject(function(elementRegistry) {
+
+      // given
+      const task = elementRegistry.get('ServiceTask_1');
+
+      // when
+      openPopup(task);
+
+      // then
+      expect(queryTemplateEntries()).to.have.length(5);
+
+    }));
+
+
+    it('should not display element templates that do not apply', inject(function(elementRegistry) {
+
+      // given
+      const task = elementRegistry.get('Task_1');
+
+      // when
+      openPopup(task);
+
+      // then
+      expect(queryTemplateEntries()).to.have.length(4);
+
+    }));
+
+
+    it('should not display applied element template', inject(function(elementRegistry, elementTemplates) {
+
+      // given
+      const task = elementRegistry.get('ServiceTask_1');
+      elementTemplates.applyTemplate(task, templates[0]);
+
+      // when
+      openPopup(task);
+
+      // then
+      expect(queryTemplateEntries()).to.have.length(4);
+
+    }));
+
+
+    it('display options to reset to plain element in correct order', inject(function(elementRegistry, elementTemplates) {
+
+      // given
+      const task = elementRegistry.get('ServiceTask_1');
+      elementTemplates.applyTemplate(task, templates[templates[0]]);
+
+      // when
+      openPopup(task);
+
+      // then
+      const entries = Array.from(domQueryAll('.djs-popup-body .entry', getMenuContainer()));
```

Comment:
My proposal is: Let us not assert the DOM structure here, but instead verify which entries the popup menu currently holds (https://github.com/bpmn-io/diagram-js/blob/develop/lib/features/popup-menu/PopupMenu.js#L92).

This way we can build much more stable tests (and also assert that we plug-in correctly via the API). Testing on DOM level is not needed from my point of view.